### PR TITLE
Improvements to `Sequence.switch_device()`

### PR DESCRIPTION
**Pull request by HGSilveri**
- Stops enforcing that every channel has an EOM config, only those that are using it
- Also checks for the channel type when matching channels

Further updates can be made with the changes in #558 , I left them as a TODO here. 
I'm also torn as to whether I release this in a patch or not... Wdyt @a-corni ?

### CODE REVIEW

**Review by a-corni — COMMENTED** [review]
I am suggesting only minor changes, otherwise it looks good to me :)
I am ok with the TODOs, I would add a TODO to check that the RegisterLayout of the Sequence's Register (if old device is a Device)/the Register (if old device is a VirtualDevice) of the Sequence fits within a RegisterLayout of pre_calibrated_layouts if the new device is a Device and has pre_calibrated_layouts (at the moment it seems to me that we check the filling, the radial distance, ... but not that the patterns match).

File: pulser-core/pulser/sequence/sequence.py [hunk 1]
```diff
@@ -509,27 +514,31 @@ def check_retarget(ch_obj: Channel) -> bool:
 
                 # We verify the channel class then
                 # check whether the addressing is Global or Local
+                type_match = type(old_ch_obj) == type(new_ch_obj)
```
[the remark this answers is not in the document]
Comment (HGSilveri) [reply pulser-core/pulser/sequence/sequence.py:517]: Any reason why you think it's preferable over this?

File: pulser-core/pulser/sequence/sequence.py [hunk 2]
```diff
@@ -509,27 +514,31 @@ def check_retarget(ch_obj: Channel) -> bool:
...
+                if old_ch_name in active_eom_channels:
                     # Uses EOM mode, so the new device needs a matching
                     # EOM configuration
                     if new_ch_obj.eom_config is None:
                         ch_match_err = base_msg + " with an EOM configuration."
                         continue
                     if (
+                        # TODO: Improvements to this check:
+                        # 1. multiple_beam_control doesn't matter when there
+                        # is only one beam
```
[the remark this answers is not in the document]
Comment (HGSilveri) [reply pulser-core/pulser/sequence/sequence.py:539]: The problem is that we can only know what beams are used when the sequence is parametrized...

File: pulser-core/pulser/sequence/sequence.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (a-corni) [reply pulser-core/pulser/sequence/sequence.py:517]: Readability could be one, I liked `old_ch_obj.name == new_ch_obj.name`, but otherwise no not really 😅

**Review by a-corni — APPROVED** [review]

**Pull request merged**